packages/jaeger-ui/src/model/trace-viewer.tsx

```diff
@@ -12,8 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { Span } from '../types/trace';
+
 // eslint-disable-next-line import/prefer-default-export
-export function getTraceName(spans) {
-  const span = spans.filter(sp => !sp.references || !sp.references.length)[0];
-  return span ? `${span.process.serviceName}: ${span.operationName}` : '';
+export function getTraceName(spans: Span[]) {
+  const allSpanIDs = spans.map(sp => sp.spanID);
+  const rootSpan = spans
+    .filter(sp => {
+      if (!sp.references || !sp.references.length) {
+        return true;
+      }
+      const parentIDs = sp.references.filter(r => r.refType === 'CHILD_OF').map(r => r.spanID);
+
+      // no parent from trace
+      return !parentIDs.some(pID => allSpanIDs.includes(pID));
+    })
+    .sort((sp1, sp2) => sp1.startTime - sp2.startTime)[0];
```

Comment:
I missed it.

[Added](https://github.com/jaegertracing/jaeger-ui/pull/541/commits/c914bcfcb199c0c4433e03081777b31d88b92717) a parents check to a `sort()` method (also, left untouched comparing by a `startTime` since, I think, it's better than not to sort multiple "root" spans at all if there are several of them).

Also, [changed](https://github.com/jaegertracing/jaeger-ui/pull/541/commits/a6f67ba80ada6ef9f65dfe9201bd2952e4790a48) a couple of tests.